app/src/main/java/org/fossasia/pslab/activity/WaveGeneratorActivity.java

```diff
@@ -621,4 +652,53 @@ private void enableInitialState() {
         activePwmPinTv = pwmMonSqr1;
         selectBtn(SQ1);
     }
+
+    private void setUpBottomSheet() {
+        bottomSheetBehavior = BottomSheetBehavior.from(bottomSheet);
+
+        final SharedPreferences settings = this.getSharedPreferences(PREFS_NAME, MODE_PRIVATE);
+        Boolean isFirstTime = settings.getBoolean("WaveGeneratorSkip", true);
+
+        bottomSheetGuideTitle.setText(R.string.wave_generator);
+        bottomSheetText.setText(R.string.wave_gen_guide_intro);
+        bottomSheetSchematic.setImageResource(R.drawable.sin_wave_guide);
+        bottomSheetDesc.setText(R.string.wave_gen_guide_desc);
+
+        if (isFirstTime) {
+            bottomSheetBehavior.setState(BottomSheetBehavior.STATE_EXPANDED);
+            tvShadow.setAlpha(0.8f);
+            arrowUpDown.setRotation(180);
+            bottomSheetSlideText.setText(R.string.hide_guide_text);
+            SharedPreferences.Editor editor = settings.edit();
+            editor.putBoolean("WaveGeneratorSkip", false);
+            editor.apply();
+        } else {
+            bottomSheetBehavior.setState(BottomSheetBehavior.STATE_HIDDEN);
+        }
+
+        bottomSheetBehavior.setBottomSheetCallback(new BottomSheetBehavior.BottomSheetCallback() {
+            @Override
```

Comment:
I have a suggestion. After a timeout, you can hide the Show guide bar. Or else you have to slide down twice to completely hide it

Reply:
nice suggestion working on it